.github/workflows/ci.yaml

```diff
@@ -87,23 +88,49 @@ jobs:
     env:
       GOPATH: /home/runner/go
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout code
+        uses: actions/checkout@v3
       - name: Setup Golang
         uses: actions/setup-go@v3.2.0
         with:
           go-version: "1.19"
-      - name: Restore Go build cache
+      - name: Restore go build cache
         uses: actions/cache@v3
         with:
-          path: ~/.cache/go-build
-          key: ${{ runner.os }}-go-build-v1-${{ github.run_id }}
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
       - run: make lint
       - run: git diff --exit-code
 
+  build-image:
```

Comment:
I pulled this out into an independent step so it can be executed once rather than for each item in the e2e-tests matrix.

Reply:
The idea is we also want to make it one step for local e2e testing, so I'm okay with each matrix testing to build the image for testing, even though it adds several minutes.